gprofiler/usage_loggers.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import logging
+import os
+import time
+from pathlib import Path
+from typing import Optional, Tuple
+
+import psutil
+
+CGROUPFS_ROOT = "/sys/fs/cgroup"  # TODO extract from /proc/mounts, this may change
+
+
+class UsageLoggerInterface:
+    def init_cycles(self):
+        raise NotImplementedError
+
+    def log_cycle(self):
+        raise NotImplementedError
+
+    def log_run(self):
+        raise NotImplementedError
+
+
+class CpuUsageLogger(UsageLoggerInterface):
+    NSEC_PER_SEC = 1000000000
+
+    def __init__(self, logger: logging.LoggerAdapter, cgroup: str):
+        self._logger = logger
+        self._cpuacct_usage = Path(f"{CGROUPFS_ROOT}{cgroup}cpuacct/cpuacct.usage")
```

Comment:
Hmm assert that `cgroup` starts with a `/`?

Reply:
Yup